File: ios/kyber/SwiftKyberModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SwiftKyberModule, NSObject)
RCT_EXTERN_METHOD(performOperation:(NSString *)data resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/SwiftKyberModule.swift
//
//  SwiftKyberModule.swift
//  Internxt
//

import Foundation
import React
import SwiftKyber

@objc(SwiftKyberModule)
class SwiftKyberModule: NSObject {
    @objc
    func performOperation(_ data: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {

        do {
            let result = try Kyber.performOperation(input: data)
            resolver(result)
        } catch {
            rejecter("ERROR", "Error usando SwiftKyber", error)
        }
    }
}
